### PR TITLE
React capstone project - Metrics web app

### PR DESCRIPTION
**Pull request by AlselwiAisha**
**Description:**
- Use React documentation.
- Use React components.
- Use React props.
- Use React Router.
- Connect React and Redux.
- Handle events in a React app.
- Write integration tests with a React testing library.
- Use styles in a React app.
- Use React life cycle methods.
- Apply React best practices and language style guides in code.
- Use store, actions and reducers in React.
- Select the API.
- Build files structure for my React app.
- Prepare routes and navigation in your app.
- User can display a list of items and filter them. Data come from the API.
- Could you create the tests for the application?
- Style my components to match the design provided.
- Deploy the project [Live Demo](https://countries-info-irzg.onrender.com/) and test for final details.
- Record a video for my project.
- Create a good README.

### CODE REVIEW

File: src/components/Navigation.jsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    <div className="navbar">
+      <NavLink to="/">
+        <IoIosArrowBack className="icon txtwhite" />
+      </NavLink>
+      <h1 className="txtwhite">Countries Info 2023</h1>
+      <NavLink to="/country">
+        <IoIosSettings className="icon txtwhite" />
+      </NavLink>
+    </div>
+  </nav>
```
Comment (shahadat3669) [inline src/components/Navigation.jsx]: - [x] Great job on adding the right font family, but in comparison to the design, the title looks a bit too large. Please consider updating the font size to better match the design.

- [x] Currently, both the home page and the details page display the same title. However, as per the design, they should change based on the page. Please update your code to display the appropriate titles for each page.

- [x] In the design, to the right side of the title, there were two icons, but you've added only one icon. Kindly try to update your code to include the missing icon and better align with the design.

| Page | Your | Design |
| --- | --- | --- | 
|  Home   |  image| image|
|  Details |  image| image|

File: src/components/HomePage.jsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            {' '}
+            Map
+          </h2>
+          <p>
+            {countriesData ? countriesData.length : countries.length}
+            {' '}
+            Countries
+          </p>
+        </div>
+      </div>
```
Comment (shahadat3669) [inline src/components/HomePage.jsx:86]: - [x] Well done on successfully changing the image and text based on the filter. However, I recommend following the design more closely. Currently, the image occupies 2/3 of the parent width, while the text takes up 1/3. As per the project requirements, each should occupy half of the width. Please consider updating your code to align with the design, ensuring both the image and text have an equal width. 👍 

| Your | Design |
| --- | --- |
|  image| image|

File: src/components/Country.jsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            {region}
+          </p>
+        </div>
+      </div>
+      <div className="details-btn">
+        <Link to={`/country/${cca2}`} onClick={() => dispatch(getCountry(commonName))}>
+          <HiOutlineArrowCircleRight className="txtwhite" />
+        </Link>
+      </div>
+    </div>
```
Comment (shahadat3669) [inline src/components/Country.jsx]: - [x] Currently, you are displaying the country name and region. However, as per the design, you should also show some numeric values alongside the country. Please consider updating your code to meet the project requirements.
- [x] The title appears to have a larger font size than the design. Kindly reduce the size to match the design specifications.
- [x] Additionally, the text should be right-aligned, but it is currently left-aligned. Please adjust the alignment accordingly.
- [x] The `nth-5` and `nth-6` cards should have different colors. Please consider adding the correct colors to these cards.
- [x] You are currently using a `div` tag to showcase the cards. As a best practice, it's recommended to use the `article` tag. Please consider updating it for better semantic structure.

| Your | Design |
| --- | --- |
|  image| image|

File: src/components/CountryDetails.jsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+const CountryDetails = () => {
+  const { isLoading, country } = useSelector((state) => state.country);
+  const countryDetails = country;
+  if (isLoading) {
+    return (
+      <div className="details" data-testid="details-test">
+        <div className="country-map">
+          <img src={countryDetails.flag} alt="Country Map" />
+        </div>
```
Comment (shahadat3669) [inline src/components/CountryDetails.jsx]: - [x] You've included an image in the headline section, which is great. However, according to the design, there should also be text accompanying the image based on countries. Please consider following the design and updating your code to include the necessary text.

| Your | Design |
| --- | --- |
|  image| image|

File: src/components/CountryDetails.jsx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+          </div>
+          <div className="item">
+            <h2>
+              Population:
+              <span>{countryDetails.population}</span>
+            </h2>
+            <HiOutlineArrowCircleRight className="icon1 txtwhite" />
+          </div>
+        </div>
+      </div>
```
Comment (shahadat3669) [inline src/components/CountryDetails.jsx]: - [x] Great job showcasing the country details. However, the spacing and text alignment do not follow the design. Each value should be right-aligned. Additionally, the left and right spacing is too much compared to the design. Please consider updating your code to align with the design.

| Your | Design |
| --- | --- |
|  image| image|

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @AlselwiAisha, 

Great job so far! 👏 You've done well, and you're almost there for the next project ⏭️, but there are a few issues 🐛 that you still need to work on.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Video presentation
Great job on your project presentation video! You adhered to the time limit, keeping it within the 5-minute mark, which is commendable. Your use of professional and clear English was excellent, making it easy to understand the project's topic and purpose. 

Your demo of the project was comprehensive and effectively highlighted all the essential features. This gave a clear sense of how the project works, which is crucial for the audience's understanding. Your choice to discuss a piece of code you found interesting or were proud of was a nice touch, as it provides insight into your thought process and enthusiasm for the project.

Overall, you did a fantastic job creating a well-structured and engaging project presentation video. Keep up the excellent work, and continue refining your presentation skills!

## Highlights
✅ The live version working fine.
✅ The project contains a `Home` page and a `Detail` page
✅ Home page filter working fine.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @AlselwiAisha,

✔️ Nice implementation of search functionality and design

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**